pkg/nfd-master/nfd-master.go

```diff
@@ -605,20 +604,81 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// validateExtendedResources validates extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) validateExtendedResources(features *nfdv1alpha1.Features, extendedResources ExtendedResources) ExtendedResources {
+	outExtendedResources := ExtendedResources{}
+	deniedNs := map[string]struct{}{"kubernetes.io": {}}
+	deniedWildCarNs := map[string]struct{}{".kubernetes.io": {}}
+	allowedNS := map[string]struct{}{nfdv1alpha1.ExtendedResourceNs: {},
+		nfdv1alpha1.FeatureLabelNs: {}}
+	for extendedResource, capacity := range extendedResources {
+		// Add possibly missing default ns
+		extendedResource = addNs(extendedResource, nfdv1alpha1.ExtendedResourceNs)
+		// Check if given NS is allowed
+		ns, _ := splitNs(extendedResource)
+		if isNamespaceDenied(ns, deniedWildCarNs, deniedNs) {
+			if _, ok := allowedNS[ns]; !ok {
+				klog.Errorf("Namespace %q is not allowed. Ignoring Extended Resource  %q\n", ns, extendedResource)
+				continue
+			}
+		}
+
+		// Dynamic Value
+		if strings.HasPrefix(capacity, "@") {
+			// Remove the @ prefix
+			capacity = strings.TrimPrefix(capacity, "@")
+			// capacity is a string in the form of attribute.featureset.elements
+			split := strings.SplitN(capacity, ".", 3)
+			if attrFeatureSet, ok := features.Attributes[split[0]+"."+split[1]]; ok {
+				if element, ok := attrFeatureSet.Elements[split[2]]; ok {
+					q, err := k8sQuantity.ParseQuantity(element)
```

Comment:
nit: I think this is a matter of style but somehow this deep nested if structure with error cases very far below is a bit hard to follow. I'd suggest to handle the error case directly after the operation, i.e. smth like

```suggestion
			featureName := split[0]+"."+split[1]
			elementName := split[2]
			attrFeatureSet, ok := features.Attributes[featureName]
			if !ok {
				klog.Errorf("feature %s not found. Ignoring Extended Resource %q", featureName, extendedResource)
				continue
			}
			element, ok := attrFeatureSet.Elements[elementName]
			if !ok {
				klog.Errorf("element %s not foundon feature %s. Ignoring Extended Resource %q", elementName, split[0]+"."+split[1], extendedResource)
				continue
			}
			q, err := k8sQuantity.ParseQuantity(element)
```


WDYT?